### ISSUE TITLE
Make grepFilterSpecs work when specPattern has been set in the config

### ISSUE THREAD

**Issue by jablesyeah** [cypress-io/cypress#24452]
The grepFilterSpecs only appears to work correctly if the specPattern value is the default value. If the value is changed, by setting it within the config, the grepFilterSpecs no longer works.

example of config:

```js
const { defineConfig } = require('cypress');

module.exports = defineConfig({
    e2e: {
        setupNodeEvents(on, config) {
            require('cypress-grep/src/plugin')(config);
            return config;
        },
        baseUrl: 'http://localhost:802/',
        specPattern: 'tests/cypress/e2e/**',
        supportFile: 'tests/cypress/support/e2e.js',
        env: {
            grepFilterSpecs: true,
            grepOmitFiltered: true,
        },
    },
});
```

With specPattern in config:
Specs:          54 found 

Without specPattern in config:
Specs:          17 found

**Comment by jablesyeah** [comment]
@s4ms00n9 @bahmutov @joshuajtward 
That's right. It appears to be related to the location of the config file.

**Comment by jablesyeah** [comment]
@joshuajtward Now we have found what this issue is, do you think it's something that can be fixed?

**Comment by joshuajtward** [comment]
Hi @jablesyeah! I can investigate this, but from the surface level it's a bit of an odd one - the way this part of the plugin works is that the resolved Cypress config is passed to the plugin (via `require('cypress-grep/src/plugin')(config)`) so I wouldn't expect the location of the config file to make a difference

Asking out of ignorance - is there a specific reason to have the config file in a different location?

I'll see if I can carve out some time this weekend to have a dig around...

**Comment by jablesyeah** [comment]
@joshuajtward Awesome. I appreciate you looking into this.
 
 I believe its something to do with our docker images etc but is an implementation made by the devs, so I'm not 100% sure.

**Comment by s4ms00n9** [comment]
> Hi @jablesyeah! I can investigate this, but from the surface level it's a bit of an odd one - the way this part of the plugin works is that the resolved Cypress config is passed to the plugin (via `require('cypress-grep/src/plugin')(config)`) so I wouldn't expect the location of the config file to make a difference
> 
> Asking out of ignorance - is there a specific reason to have the config file in a different location?
> 
> I'll see if I can carve out some time this weekend to have a dig around...

My case is that I have several config files for different environments and I store them in a separate folder (which makes the location not default).
And maybe as a hint - it is not mentioned in the official docs, but I think adding `grepIntegrationFolder: "../../"` (pointing to a root folder from where the config file is) into env may be a workaround for this issue.

**Comment by joshuajtward** [comment]
Yeah good point @s4ms00n9, `grepIntegrationFolder` is a bit of an artefact from when I was doing the upgrade to work with `10.0.0`, I initially wrote it to be backward compatible with `9.0.0`, but I was persuaded to make it a breaking change to gently push people to upgrade. I think that's why I forgot to add any info to the README about it 🤦‍♂️

For this issue going forward, I think `grepIntegrationFolder` should probably be renamed to something like `relativeConfigFileToIntegrationFolderPath` (or use `path` to resolve two paths for us), and I should add something to the docs for it. @jablesyeah does setting it to be the relative path for your setup work? (not as a solution, but to validate that we're on the right track) 

In addition, I've just seen this: cypress-io/cypress-grep#153, so we should probably move this issue to the main Cypress repo, as `cypress-grep` excitingly now lives here: https://github.com/cypress-io/cypress/tree/develop/npm/grep
